include/Task.hpp

```diff
@@ -0,0 +1,15 @@
+//
+// Created by Tianyi Wang on 5/14/23.
+//
+
+#ifndef DD_EVAL_TASK_HPP
+#define DD_EVAL_TASK_HPP
+
+#endif // DD_EVAL_TASK_HPP
+
+#include "Result.hpp"
+
+class Task {
+public:
+  virtual Result execute() = 0;
```

Comment:
We'll have to see if that signature works as this PR develops. There could be a problem with object slicing due to the return type here being `Result`. If you return an object of a class derived from `Result` this will most likely not work.
If I had a guess, a signature like
```c++
virtual json execute() = 0;
```
could work best, that more or less gets rid of the Results class.